lib/migration_tools/tasks.rb

```diff
@@ -27,47 +30,131 @@ def migrations_paths
+      @multi_database_setup ||= if ActiveRecord::VERSION::MAJOR >= 6
+                                  database_configs_hash.reject { |c| c.spec_name.end_with?('_replica') }.size > 1
+                                else
+                                  false
+                                end
+    end
+
     def migrator(target_version = nil)
       if ActiveRecord::VERSION::MAJOR >= 6
         migrate_up(ActiveRecord::MigrationContext.new(
           migrations_paths,
           ActiveRecord::SchemaMigration
-        ).migrations, target_version)
+        ).migrations, ActiveRecord::SchemaMigration, target_version)
       elsif ActiveRecord::VERSION::MAJOR == 5 && ActiveRecord::VERSION::MINOR == 2
         migrate_up(ActiveRecord::MigrationContext.new(migrations_paths).migrations, target_version)
       else
         migrate_up(ActiveRecord::Migrator.migrations(migrations_paths), target_version)
       end
     end
 
-    def migrate_up(migrations, target_version)
-      if ActiveRecord::VERSION::MAJOR  >= 6
-        ActiveRecord::Migrator.new(:up, migrations, ActiveRecord::SchemaMigration, target_version)
+    def migrate_up(migrations, schema_migration, target_version)
+      if ActiveRecord::VERSION::MAJOR >= 6
+        ActiveRecord::Migrator.new(:up, migrations, schema_migration, target_version)
       else
         ActiveRecord::Migrator.new(:up, migrations, target_version)
       end
     end
 
+    def migrator_for_multi_database(db_config, target_version)
+      connection = ActiveRecord::Base.establish_connection(db_config.config).connection
+
+      migrations_paths = db_config.config[:migrations_paths] || db_config.config['migrations_paths'] || 'db/migrate'
+
+      migrations_paths = Array(migrations_paths)
+
+      migrations = ActiveRecord::MigrationContext.new(migrations_paths, connection.schema_migration).migrations
+
+      migrate_up(migrations, connection.schema_migration, target_version)
+    end
+
+    def multi_db_pending_migrations
+      return @multi_db_pending_migrations if defined?(@multi_db_pending_migrations) && @multi_db_pending_migrations
+
+      @multi_db_pending_migrations ||= database_configs_hash.each_with_object({}) do |db_config, hash|
+        single_db_migrator = migrator_for_multi_database(db_config, nil)
+        hash[db_config.spec_name] = {
+          pending_migrations: filter_pending_migrations_for_group(single_db_migrator.pending_migrations),
+          db_config: db_config
+        }
+      end
+
+      @multi_db_pending_migrations
+    end
+
     def pending_migrations
+      return [] if multi_database_setup?
       return @pending_migrations if defined?(@pending_migrations) && @pending_migrations
-      @pending_migrations = migrator.pending_migrations
-      @pending_migrations = @pending_migrations.select { |proxy| group.empty? || proxy.migration_group == group }
+
+      @pending_migrations = filter_pending_migrations_for_group(migrator.pending_migrations)
```

Comment:
Same here. If you used the conditional assignment operator `||=` you probably wouldn't need line 94